### PR TITLE
Upgrade Prometheus Operator to 0.38.1

### PR DESCRIPTION
**Pull request by omegas27**
Towards: https://github.com/giantswarm/giantswarm/issues/10897

- Upgrading the Prometheus operator and dashboards to the latest 0.39.0 version
- Update the CI job to deploy Prometheus operator to the control plane catalog
- Remove the deleted experimental crdApiGroup

I will test once the quay situation is resolved

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,12 +27,11 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 - Update kube-state-metrics chart to 2.4.1
 - Update prometheus-node-exporter chart to 1.8.1
 
-
 ## [v0.1.1]
 
 ### Changed
 
-- Disable App Armor in grafana as it does not work on CoreOS.
+- Disable App Armor in grafana as it does not work on CoreOS and Flatcar.
```
Comment (pipo02mix) [inline CHANGELOG.md:34]: <3

File: helm/prometheus-operator-app-chart/ci/02-test-without-crds-values.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+prometheusOperator:
+  createCustomResource: false
+  namespaces:
+    releaseNamespace: true
+    additional:
+      - kube-system
```
Comment (pipo02mix) [inline helm/prometheus-operator-app-chart/ci/02-test-without-crds-values.yaml:6]:
```suggestion
      - kube-system

```

File: helm/prometheus-operator-app-chart/ci/01-provision-crds-values.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    selfMonitor: false
+  createCustomResource: false
+  tlsProxy:
+    enabled: false
+  admissionWebhooks:
+    enabled: false
+  namespaces:
+    releaseNamespace: true
+    additional:
+      - kube-system
```
Comment (pipo02mix) [inline helm/prometheus-operator-app-chart/ci/01-provision-crds-values.yaml:42]:
```suggestion
      - kube-system

```

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,16 +1,46 @@
...
+              only: /^v.*/
+
+      - architect/push-to-app-catalog:
+          name: push-to-app-catalog-pr
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "prometheus-operator-app-chart"
+          filters:
+            branches:
+              ignore: master
```
Comment (TheoBrigitte) [inline .circleci/config.yml]: Having a single job to push to `control-plane-catalog` is enough.

By default circleci builds for every branches but not tag, so let's keep one job which is triggered for every branches and every tag.

i.e. 
```
          filters:
            tags:
              only: /^v.*/
```

File: helm/prometheus-operator-app-chart/Chart.yaml [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
 name: prometheus-operator-app-chart
-sources:
-- https://github.com/giantswarm/prometheus-operator-app
+namespace: monitoring
 version: [[ .Version ]]
+sources:
+- https://raw.githubusercontent.com/giantswarm/prometheus-operator-app/v[[ .Version ]]/README.md
+- https://github.com/coreos/kube-prometheus
+- https://github.com/coreos/prometheus-operator
+- https://coreos.com/operators/prometheus
```
Comment (TheoBrigitte) [inline helm/prometheus-operator-app-chart/Chart.yaml:13]:
```suggestion
- https://coreos.com/operators/prometheus

```

**Comment by omegas27** [comment]
Diff:
```bash
diff  charts/stable/prometheus-operator/ prometheus-operator-app/helm/prometheus-operator-app-chart
Only in prometheus-operator-app/helm/prometheus-operator-app-chart: charts
diff charts/stable/prometheus-operator/Chart.yaml prometheus-operator-app/helm/prometheus-operator-app-chart/Chart.yaml
1a2
> appVersion: v0.39.0
2a4
> home: https://github.com/giantswarm/prometheus-operator-app
4,10c6,8
< engine: gotpl
< maintainers:
<   - name: vsliouniaev
<   - name: bismarck
<   - name: gianrubio
<     email: gianrubio@gmail.com
< name: prometheus-operator
---
> name: prometheus-operator-app-chart
> namespace: monitoring
> version: [[ .Version ]]
12,21c10,13
<   - https://github.com/coreos/kube-prometheus
<   - https://github.com/coreos/prometheus-operator
<   - https://coreos.com/operators/prometheus
< version: 8.13.8
< appVersion: 0.38.1
< tillerVersion: ">=2.12.0"
< home: https://github.com/coreos/prometheus-operator
< keywords:
< - operator
< - prometheus
---
> - https://raw.githubusercontent.com/giantswarm/prometheus-operator-app/v[[ .Version ]]/README.md
> - https://github.com/coreos/kube-prometheus
> - https://github.com/coreos/prometheus-operator
> - https://coreos.com/operators/prometheus
\ No newline at end of file
Common subdirectories: charts/stable/prometheus-operator/ci and prometheus-operator-app/helm/prometheus-operator-app-chart/ci
Only in charts/stable/prometheus-operator/: CONTRIBUTING.md
Common subdirectories: charts/stable/prometheus-operator/crds and prometheus-operator-app/helm/prometheus-operator-app-chart/crds
Common subdirectories: charts/stable/prometheus-operator/hack and prometheus-operator-app/helm/prometheus-operator-app-chart/hack
Only in charts/stable/prometheus-operator/: OWNERS
Only in charts/stable/prometheus-operator/: README.md
diff charts/stable/prometheus-operator/requirements.lock prometheus-operator-app/helm/prometheus-operator-app-chart/requirements.lock
4c4
<   version: 2.8.4
---
>   version: 2.8.5
10,12c10,12
<   version: 5.0.18
< digest: sha256:ae93b2266b7083b6b5d74b0fd5e224a93572766b97e1ba74bca0e6f3c69ed094
< generated: "2020-04-29T11:40:21.286237214+02:00"
---
>   version: 5.0.25
> digest: sha256:1dc755d615f6d95bf38ba872cf8794c8ebf29ac46a4d5beb5b50bab80189ee1b
> generated: "2020-05-18T17:03:28.527893017+02:00"
diff charts/stable/prometheus-operator/requirements.yaml prometheus-operator-app/helm/prometheus-operator-app-chart/requirements.yaml
4c4
<     version: "2.8.*"
---
>     version: 2.8.*
9c9
<     version: "1.9.*"
---
>     version: 1.9.*
14c14
<     version: "5.0.*"
---
>     version: 5.0.*
16c16
<     condition: grafana.enabled
---
>     condition: grafana.enabled
\ No newline at end of file
Common subdirectories: charts/stable/prometheus-operator/templates and prometheus-operator-app/helm/prometheus-operator-app-chart/templates
diff charts/stable/prometheus-operator/values.yaml prometheus-operator-app/helm/prometheus-operator-app-chart/values.yaml
325c325
<       repository: quay.io/prometheus/alertmanager
---
>       repository: quay.io/giantswarm/alertmanager
480a481,483
>   rbac:
>     pspUseAppArmor: false
> 
1076c1079
<       repository: squareup/ghostunnel
---
>       repository: quay.io/giantswarm/ghostunnel
1093c1096
<         repository: jettech/kube-webhook-certgen
---
>         repository: quay.io/giantswarm/kube-webhook-certgen
1262,1263c1265,1266
<     repository: quay.io/coreos/prometheus-operator
<     tag: v0.38.1
---
>     repository: quay.io/giantswarm/prometheus-operator
>     tag: v0.39.0
1269c1272
<     repository: quay.io/coreos/configmap-reload
---
>     repository: quay.io/giantswarm/configmap-reload
1275,1276c1278,1279
<     repository: quay.io/coreos/prometheus-config-reloader
<     tag: v0.38.1
---
>     repository: quay.io/giantswarm/prometheus-config-reloader
>     tag: v0.39.0
1289,1290c1292,1293
<     repository: k8s.gcr.io/hyperkube
<     tag: v1.12.1
---
>     repository: quay.io/giantswarm/hyperkube
>     tag: v1.17.5
1509c1512
<       repository: quay.io/prometheus/prometheus
---
>       repository: quay.io/giantswarm/prometheus

```

**Comment by omegas27** [comment]
Closed in favor of https://github.com/giantswarm/prometheus-operator-app/pull/22

**Pull request closed without merging**